lib/manager/util/validation.js

```diff
@@ -0,0 +1,10 @@
+// @flow
+
+/**
+ * Checks if a filename is valid.
+ * A valid filename does not contain <>:"/\|?* characters and does not end with .zip (case-insensitive).
+ * An empty or null/undefined filename is also considered valid.
+ */
+export function isValidFilename (filename: ?string): boolean {
+  return !filename || (/^[^<>:"/\\|?* ]*$/.test(filename) && !/\.zip$/i.test(filename))
```

Comment:
Can we also test if it ends in a special character and can we have it check for `.zip` at any point? 
<img width="224" alt="Screenshot 2025-04-30 at 4 33 46 PM" src="https://github.com/user-attachments/assets/9eca284f-89e5-439a-8823-b57b02b5d0bf" />